core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.spec.ts

```diff
@@ -0,0 +1,226 @@
+    version: 1,
+    misconception_count: null,
+    worked_examples_count: null,
+    skill_model_created_on: 2,
+    skill_model_last_updated: 3
+  };
+
+  let CategorizedSkillsDict: {
+    test: {
+      uncategorized: [],
+      test: []
+    }
+  };
+
+  let skillIdToRubricsObject = {};
+
+  let UntriagedSkillSummaries: SkillSummary[] = (
+    [SkillSummary.createFromBackendDict(skillSummaryBackendDict)]);
+
+  const topicsAndSkillsDashboardData: TopicsAndSkillDashboardData = {
+    allClassroomNames: null,
+    canDeleteTopic: null,
+    canCreateTopic: null,
+    canDeleteSkill: null,
+    canCreateSkill: null,
+    untriagedSkillSummaries: UntriagedSkillSummaries,
+    mergeableSkillSummaries: null,
+    totalSkillCount: null,
+    topicSummaries: null,
+    categorizedSkillsDict: CategorizedSkillsDict
+  };
+
+  class MockTopicsAndSkillsDashboardBackendApiService {
+    success: boolean = true;
+    fetchDashboardDataAsync() {
+      return {
+        then: (callback: (resp) => void) => {
+          callback(topicsAndSkillsDashboardData);
+        }
+      };
+    }
+  }
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        TopicsAndSkillsDashboardBackendApiService,
+        {
+          provide: TopicsAndSkillsDashboardBackendApiService,
+          useClass: MockTopicsAndSkillsDashboardBackendApiService
+        }
+      ]
+    });
+    qls = TestBed.get(QuestionsListService);
+    focusManagerService = TestBed.get(FocusManagerService);
+  });
+
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $window = $injector.get('$window');
+    TopicEditorStateService = $injector.get('TopicEditorStateService');
+    TopicObjectFactory = $injector.get('TopicObjectFactory');
+    focusManagerService = $injector.get('FocusManagerService');
+    topicInitializedEventEmitter = new EventEmitter();
+    topicReinitializedEventEmitter = new EventEmitter();
+
+    ctrl = $componentController('questionsTab', {
+      $scope: $scope,
+    });
+
+    topic = TopicObjectFactory.createInterstitialTopic();
+    subtopic1 = Subtopic.createFromTitle(1, 'Subtopic1');
+    subtopic1.addSkill('skill1', 'subtopic1 skill');
+    topic.getSubtopics = function() {
+      return [subtopic1];
+    };
+
+    spyOn(TopicEditorStateService, 'getTopic').and.returnValue(topic);
+    spyOn(focusManagerService, 'setFocus');
+    spyOnProperty(TopicEditorStateService, 'onTopicInitialized').and.callFake(
+      function() {
+        return topicInitializedEventEmitter;
+      });
+    spyOnProperty(
+      TopicEditorStateService, 'onTopicReinitialized').and.callFake(
+      function() {
+        return topicReinitializedEventEmitter;
+      });
+    ctrl.$onInit();
+  }));
+
+  afterEach(() => {
+    ctrl.$onDestroy();
+  });
+
+  it('should initialize the variables', function() {
```

Comment:
missing the action in the name.. When should it initialize the variables?

Reply:
DONE